src/Akeneo/Category/front/src/feature/components/templates/AttributeList.tsx

```diff
@@ -0,0 +1,92 @@
+import {Button, SectionTitle, Table, useBooleanState} from 'akeneo-design-system';
+import {Attribute} from '../../models';
+import {getLabelFromAttribute} from '../attributes';
+import React, {useCallback, useState} from 'react';
+import {useFeatureFlags, userContext, useTranslate} from '@akeneo-pim-community/shared';
+import styled from 'styled-components';
+import {AddTemplateAttributeModal} from './AddTemplateAttributeModal';
+
+type Props = {
+  attributes: Attribute[];
+  templateId: string;
+  onAttributeSelection: (attribute: Attribute) => void;
+};
+
+export const AttributeList = ({attributes, templateId, onAttributeSelection}: Props) => {
+  const translate = useTranslate();
+  const catalogLocale = userContext.get('catalogLocale');
+  const featureFlags = useFeatureFlags();
+
+  const [selectedAttribute, setSelectedAttribute] = useState<Attribute>(attributes[0]);
+  const [isAddTemplateAttributeModalOpen, openAddTemplateAttributeModal, closeAddTemplateAttributeModal] =
+    useBooleanState(false);
+
+  const handleRowOnclick = (attribute: Attribute) => {
+    console.log('row clicked. attribute label=' + attribute.labels[catalogLocale]);
+    console.log('we should highlight and rerender the attribute line');
+    setSelectedAttribute(attribute);
+    onAttributeSelection(attribute);
+  };
+
+  const sortByOrder = useCallback((attribute1: Attribute, attribute2: Attribute): number => {
+    if (attribute1.order >= attribute2.order) {
+      return 1;
+    } else if (attribute1.order < attribute2.order) {
+      return -1;
+    }
+    return 0;
+  }, []);
```

Comment:
Something you can do to make it a bit more explicit is to memoize direclty the sorted list here:
```suggestion
  const sortedAttributes = useMemo(() => attributes.sort((attribute1: Attribute, attribute2: Attribute): number => {
    if (attribute1.order >= attribute2.order) {
      return 1;
    } else if (attribute1.order < attribute2.order) {
      return -1;
    }
    return 0;
  }), [attributes]);
```